src/modules/function/declaration_utils.rs

```diff
@@ -6,23 +6,32 @@ use crate::utils::cc_flags::{CCFlags, get_ccflag_name};
 use crate::utils::context::Context;
 use crate::utils::function_interface::FunctionInterface;
 
-pub fn skip_function_body(meta: &mut ParserMetadata) -> (usize, usize, bool) {
+pub fn skip_function_body(
+    meta: &mut ParserMetadata,
+    declared_failable: bool,
+    returns_tok: &Option<Token>,
+) -> Result<(usize, usize, bool), Failure> {
     let index_begin = meta.get_index();
```

Comment:
If you extract the error handling, you also won't need to pass these arguments or make the function return a `Result`.